org-ql.el

```diff
@@ -448,6 +452,29 @@ Returns cons (INHERITED-TAGS . LOCAL-TAGS)."
                  org-ql-tags-cache))
       (puthash position all-tags tags-cache))))
 
+(defun org-ql--expand-tag-groups (tags &optional exclude-groups)
+  (if (eq tags 'org-ql-nil)
+      'org-ql-nil
+    (let ((exclude-groups (append tags exclude-groups))
+          result)
+      (let (group-tags)
+        (dolist (tag tags)
+          (dolist (groups org-tag-groups-alist)
```

Comment:
This would be a good opportunity to use `pcase-dolist` to improve the code's legibility (i.e. by avoiding `(car groups)`, which doesn't tell what the CAR is).

Reply:
I personally haven't been using `pcase` much so it didn't cross my mind. I definitely should make a habit out of this. Addressed in the new commit.